ci/vulcan/home.nix

```diff
@@ -0,0 +1,37 @@
+{ config, pkgs, ... }:
+
+# Note: This file must be symlinked to the proper home-manager config directory.
+
+let
+  webring-flake = (builtins.getFlake "path:/home/ring/webring").packages.x86_64-linux;
+  webring = webring-flake.phwebring;
+  default-static = webring-flake.default-static;
+in {
+  home.username = "ring";
+  home.homeDirectory = "/home/ring";
+  home.stateVersion = "24.11";
+
+  home.packages = [ webring ];
+
+  home.file."webring-data/static".source = default-static;
+
+  # This will automatically get restarted when rebuilding the home directory
+  systemd.user.services.phwebring = {
+    Unit = {
+      Description = "Purdue Hackers webring";
+    };
+
+    Service = {
+      ExecStart = "${pkgs.bubblewrap}/bin/bwrap --bind /home/ring/webring-data /webring --ro-bind /nix/store /nix/store --ro-bind /etc /etc --tmpfs /tmp --unshare-all --share-net --new-session --chdir /webring --uid 256 --gid 512 --die-with-parent ${webring}/bin/ph-webring";
```

Comment:
Instead of using bwrap inline here, it may be worthwhile to instead use the systemd unit file policies to represent a similar set of permissions. This would reduce the complexity of the command invocation.

The options you need are mainly in the [Sandboxing section of systemd.exec](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#Sandboxing)

I will grant, I don't know if user units have restrictions on it, but I don't think it does in this case (outside of not being able to grant write access to paths it can't write to, of course.)

Reply:
There seem to be a lot of options in there and they're all allow by default. It looks like it would be a lot less config to keep the bwrap command since it's deny by default.

Reply:
Let's just stick with bwrap.